lib/routes/account.js

```diff
@@ -1723,12 +1703,56 @@ module.exports = function (
               uaOSVersion: sessionToken.uaOSVersion,
               uaDeviceType: sessionToken.uaDeviceType
             }
-          ))
+
+            return func(emails, sessionToken, mailerOpts)
+          })
           .then(() => request.emitMetricsEvent(`email.${event}.resent`))
           .then(
             () => reply({}),
             reply
           )
+
+        function setVerifyCode() {
+          return db.accountEmails(sessionToken.uid)
+            .then((emailData) => {
+              if (email) {
+                // If an email address is specified in payload, this is a request to verify
+                // a secondary email. This should return the corresponding email code for verification.
+                let emailVerified = false
+                emailData.some((userEmail) => {
+                  if (userEmail.normalizedEmail === email.toLowerCase()) {
+                    code = userEmail.emailCode
+                    emailVerified = userEmail.isVerified
+                    emails = [userEmail]
+                    return true
+                  }
+                })
+
+                // Don't resend code for already verified emails
+                if (emailVerified) {
+                  return reply({})
+                }
+              } else if (sessionToken.tokenVerificationId) {
+                emails = emailData
+                code = sessionToken.tokenVerificationId
+              } else {
+                code = sessionToken.emailCode
+              }
+            })
+        }
+
+        function setVerifyFunction() {
+          if (email) {
+            func = mailer.sendVerifySecondaryEmail
```

Comment:
Seeing as you're adding this setter function, maybe we could use it as an opportunity to improve the name of `func` here and above to something a bit more descriptive. `verifyFunction` or whatever...